python/shotgun_model/data_handler_nav.py

```diff
@@ -52,11 +53,22 @@ def __init__(self, root_path, seed_entity_field, entity_fields, cache_path):
             of field names to return.
 
         :param str cache_path: Path to cache file location.
+
+        :param str include_root: Defines the name of an additional, top-level
+            model item that represents the root. In views, this item will appear
+            as a sibling to top-level children of the root. This allows for
+            UX whereby a user can select an item representing the root without
+            having a UI that shows a single, top-level item. An example would
+            be displaying published file entity hierarchy with top level items:
+            "Assets", "Shots", and "Project Publishes". In this example, the
+            supplied arg would look like: ``include_root="Project Publishes"``.
+            If ``include_root`` is ``None``, no root item will be added.
         """
         super(ShotgunNavDataHandler, self).__init__(cache_path)
         self.__root_path = root_path
         self.__seed_entity_field = seed_entity_field
         self.__entity_fields = entity_fields
+        self.__include_root = include_root
```

Comment:
I'm fine to use the item label for display, but I think it will render this solution less effective. Again, I know this is an odd solution, but using the root entity label will add confusion I think and it will also mean we probably need to do additional work to make it show up on top.